tests/kani/Intrinsics/PtrGuaranteedCmp/ne_ok.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Checks that `ptr_guaranteed_ne` returns false when the pointers are
+// equal, which causes the test to pass in the `else` branch. The `if`
+// branch is unreachable.
```

Comment:
Should we test that this branch is unreachable, maybe with an assert false?

Reply:
Because we are assuming `ptr1 == ptr2`, the unreachable assertion is in practice `assert!(false)`.